cageseq.nf

```diff
@@ -0,0 +1,262 @@
+        meta, bam -> meta.id = meta.id.split('_')[0..-2].join('_')
+        [ meta, bam ]
+    }
+    .groupTuple(by: [0])
+    .map { it -> [ it[0], it[1].flatten() ] }
+    .set { ch_fastq }
+
+    // FASTQC
+    FASTQC( ch_fastq )
+    fastqc_html = FASTQC.out.html
+    fastqc_zip = FASTQC.out.zip
+    fastqc_version = FASTQC.out.version
+
+    // Channel for software version
+    ch_software_versions = Channel.empty()
+    ch_software_versions = ch_software_versions.mix(FASTQC.out.version.first().ifEmpty(null))
+
+
+    // Convert GTF to Bed format
+    GTF2BED( ch_gtf )
+
+    // Get chromosome sizes
+    GET_CHROM_SIZES( ch_fasta )
+    
+    // Trim adapters
+    ch_cutadapt_multiqc = Channel.empty()
+    TRIMMING_PREPROCESSING( 
+        ch_fastq,
+        ch_5end_artifacts,
+        ch_3end_artifacts
+        )
+    if (!params.skip_trimming){
+        ch_software_versions = ch_software_versions.mix(TRIMMING_PREPROCESSING.out.cutadapt_version.first().ifEmpty(null))
+        ch_cutadapt_multiqc = TRIMMING_PREPROCESSING.out.cutadapt_log
+    }
+    
+    ch_reads = TRIMMING_PREPROCESSING.out.reads
+    
+    // Removal ribosomal RNA
+    ch_sortmerna_multiqc = Channel.empty()
+    if (params.remove_ribo_rna) {
+        SORTMERNA( ch_reads, fasta_sortmerna )
+
+        ch_reads = SORTMERNA.out.reads
+        ch_sortmerna_multiqc = SORTMERNA.out.log
+        ch_software_versions = ch_software_versions.mix(SORTMERNA.out.version.first().ifEmpty(null))
+    }
+
+    // Optional post-preprocessing QC
+    ch_fastqc_post_multiqc = Channel.empty()
+    if(!params.skip_trimming_fastqc && !params.skip_trimming){
+        FASTQC_POST( ch_reads )
+        ch_fastqc_post_multiqc = FASTQC_POST.out.zip
+    }
+    
+    // Align with STAR
+    ch_samtools_stats    = Channel.empty()
+    ch_samtools_flagstat = Channel.empty()
+    ch_samtools_idxstats = Channel.empty()
+    ch_star_multiqc      = Channel.empty()
+    ch_bowtie_multiqc    = Channel.empty()
+    if (params.aligner == 'star'){
+        ALIGN_STAR(
+            ch_reads,
+            params.star_index,
+            ch_fasta,
+            ch_gtf
+            )
+        ch_bam = ALIGN_STAR.out.bam
+
+        ch_samtools_stats    = ALIGN_STAR.out.stats
+        ch_samtools_flagstat = ALIGN_STAR.out.flagstat
+        ch_samtools_idxstats = ALIGN_STAR.out.idxstats
+        ch_star_multiqc      = ALIGN_STAR.out.log_final
+        ch_software_versions = ch_software_versions.mix(ALIGN_STAR.out.star_version.first().ifEmpty(null))
+        ch_software_versions = ch_software_versions.mix(ALIGN_STAR.out.samtools_version.first().ifEmpty(null))
+    }
+    // Align with bowtie1
+    else if (params.aligner == 'bowtie1'){
+        ALIGN_BOWTIE(
+            ch_reads,
+            params.bowtie_index,
+            ch_fasta,
+            ch_gtf
+        )
+        ch_bam = ALIGN_BOWTIE.out.bam
+        // ch_software_versions = ch_software_versions.mix(ALIGN_BOWTIE.out.bowtie_version.first().ifEmpty(null))
```

Comment:
```suggestion
```

Reply:
there was a module-related issue with the version naming, fixed this now.